### PR TITLE
(Wallet) Implement new unlock screen to access Wallet

### PR DESCRIPTION
**Pull request by simoarpe**
## Resolves https://github.com/brave/brave-browser/issues/37720

This PR implements new unlock screen to access Brave Wallet following Figma design.
The main changes affect the XML layout and include new light/night colors.

🎨 Implement new unlock screen following [Figma design](https://www.figma.com/file/oXIiv45pIhF2RGlpEjOfUg/Wallet-onboarding?node-id=6223%3A70942&mode=dev).

☂ Main umbrella issue: https://github.com/brave/brave-browser/issues/35294

Other useful Figma links:
* 🤖 [Android components](https://www.figma.com/file/loFHVWcBAuuApFsH9V3T5I/%F0%9F%A4%96-Android?node-id=500%3A15201&mode=dev)
* 🦁 [Nala design system](https://www.figma.com/file/g8Z0q6TMPYDq6zXh9Y7LWD/%F0%9F%8C%8E-Universal?node-id=8323%3A14947&mode=dev)

## UI Preview

|  Light Theme  | Dark Theme  |
|---|---|
|   |   |

## Demo

https://github.com/brave/brave-core/assets/3308503/7aed180d-1ec5-411c-b18c-4df169d67259

## Reviewer's Notes

Some details affecting layout components shown in the unlock screen may change soon as they are currently discussed in the [Figma page](ttps://www.figma.com/file/oXIiv45pIhF2RGlpEjOfUg/Wallet-onboarding?node-id=6223%3A70942&mode=dev) by the Design team.

* Text selection handles use the color provided in default state. In the error state the only part transormed is the blinking cursor.
* Biometric button assets were downloaded directly from Figma but the overall effect is different from the one shown in the Figma preview.
* Some colors miss their dark counterpart and are currently under review (e.g. text subtitle color).
* Some effects have been ignored as they were iOS specific (e.g. sliding up transition to show unlock screen) 
* Face to unlock icon has not been included as currently not supported.
* Buttons and edit texts rounded corners are using Desktop values and may change soon.

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Without Biometric Support
- Create a new Wallet without biometric support
- Lock the Wallet (*)
- Navigate to Wallet section to open unlock screen
- Observe the button is disabled when text field is empty
- Observe biometric icon is not shown

### With Biometric Support
- Create a new Wallet with biometric support
- Lock the Wallet (*)
- Navigate to Wallet section to open unlock screen
- Observe the biometric dialog is shown immediately
- Tap outside the dialog
- Observe biometric icon is shown
- Tap on biometric icon
- Observe dialog is shown\

(*) To lock the Wallet tap on Wallet icon, tap on the overflow menu icon and select option to lock Wallet.

### CODE REVIEW

**Review by SergeyZhukovsky — APPROVED** [review]
great job ++

File: android/java/res/values-night/brave_colors.xml [hunk 1]
```diff
@@ -51,6 +51,14 @@
 
     <!-- Brave wallet 2.0 colors -->
     <color name="brave_wallet_dn_panel_bg">@color/wallet_bg</color>
+    <color name="brave_wallet_semantic_button_disabled">#4D687B85</color>
```
Comment (deeppandya) [inline android/java/res/values-night/brave_colors.xml:54]: can we use nala color names here ?

File: android/java/res/values/brave_colors.xml [hunk 2]
```diff
@@ -38,7 +38,15 @@
     <color name="brave_wallet_dn_button_color">#3F39E8</color>
     <color name="brave_wallet_checkbox_color">@color/brave_wallet_dn_button_color</color>
     <!-- Brave wallet 2.0 colors -->
+
     <color name="brave_wallet_dn_panel_bg">#EDEEF1</color>
+    <color name="brave_wallet_semantic_button_disabled">#4D687B85</color>
```
Comment (deeppandya) [inline android/java/res/values/brave_colors.xml:43]: can we use nala color names here ?

File: android/java/res/layout/fragment_unlock_wallet.xml [hunk 3]
```diff
@@ -1,119 +1,96 @@
...
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_horizontal"
+        android:layout_marginBottom="40dp"
+        android:background="?android:attr/selectableItemBackground"
+        android:clickable="true"
+        android:focusable="true"
+        android:padding="16dp"
+        android:text="@string/restore"
+        android:textColor="@color/brave_wallet_dn_button_color"
+        android:textStyle="bold"
```
Comment (deeppandya) [inline android/java/res/layout/fragment_unlock_wallet.xml:78]: we can use styling from nala here.

Comment (simoarpe) [reply android/java/res/layout/fragment_unlock_wallet.xml:78]: @deeppandya, where can I find a comprehensive list of the Nala styles available? 
Which is the one that matches this style? The restore button is a plain text made clickable, I'm attaching a screenshot.

File: browser/ui/android/strings/android_brave_strings.grd [hunk 4]
```diff
@@ -2015,9 +2015,6 @@ Are you sure you want to do this?
     <message name="IDS_RETYPE_PASSWORD" desc="Brave wallet password hint text">
       Verify password
     </message>
-    <message name="IDS_PASSWORD_ERROR" desc="Brave Wallet password error text">
```
Comment (mkarolin) [inline browser/ui/android/strings/android_brave_strings.grd:2018]: Was this unused?

**Review by mkarolin — APPROVED** [review]
`strings`++

File: android/java/res/layout/fragment_unlock_wallet.xml (hunk 3 above)
Comment (simoarpe) [reply android/java/res/layout/fragment_unlock_wallet.xml:78]: UPDATE: I've included the predefined text style `DefaultRegular` that sets some extra details.

File: browser/ui/android/strings/android_brave_strings.grd (hunk 4 above)
Comment (simoarpe) [reply browser/ui/android/strings/android_brave_strings.grd:2018]: Correct, it was unused.

File: android/java/res/values-night/brave_colors.xml (hunk 1 above)
Comment (simoarpe) [reply android/java/res/values-night/brave_colors.xml]: Color name updated using Nala naming convention. Done in commit 👉 [`0472a10` (#23338)](https://github.com/brave/brave-core/pull/23338/commits/0472a10d76e8db6a13ffa5079680c11d02c9fe77)

**Comment by simoarpe** [comment]
@deeppandya , ready for another review round 😁

File: android/java/res/values/brave_colors.xml [hunk 5]
```diff
@@ -38,7 +38,15 @@
     <color name="brave_wallet_dn_button_color">#3F39E8</color>
     <color name="brave_wallet_checkbox_color">@color/brave_wallet_dn_button_color</color>
     <!-- Brave wallet 2.0 colors -->
+
     <color name="brave_wallet_dn_panel_bg">#EDEEF1</color>
+    <color name="button_disabled">#4D687B85</color>
+    <color name="brave_wallet_semantic_text_disabled">#8021272A</color>
```
[the remark this answers is not in the document]
Comment (simoarpe) [reply android/java/res/values/brave_colors.xml]: Sure! Done in commit 👉 [`515a20e` (#23338)](https://github.com/brave/brave-core/pull/23338/commits/515a20e913e6db8543f99a74ce68687c5a832668)

File: android/java/res/layout/fragment_unlock_wallet.xml [hunk 6]
```diff
@@ -1,119 +1,97 @@
...
-            <ImageView
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:contentDescription="@null"
-                android:layout_gravity="center"
-                app:srcCompat="?attr/ic_onboarding_password"/>
-
-        </FrameLayout>
+    <TextView
+        android:id="@+id/unlock_wallet_subtitle"
```
[the remark this answers is not in the document]
Comment (simoarpe) [reply android/java/res/layout/fragment_unlock_wallet.xml:24]: Done in commit 👉 [`a52cd41` (#23338)](https://github.com/brave/brave-core/pull/23338/commits/a52cd412748ffcb17ebe9e7db48021e3c4ee7330)

**Comment by simoarpe** [comment]
@deeppandya there is only a text view in this PR that does not reflect Nala styles and uses the old style `BraveWalletTextViewTitle`. I'm going to take care of that in a separate PR where I'll change the occurrences all together. 
Let me know if you can spot anything else.

**Review by deeppandya — APPROVED** [review]
LGTM

**Comment by deeppandya** [comment]
Thank you @simoarpe for the changes related to nala tokens here.

**Pull request merged**